src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -2735,25 +2691,8 @@ static void Main()
 ";
 
             var comp = CreateCompilationWithMscorlib(source, references: s_valueTupleRefs);
+            // this is permitted now, as it is just an assignment expression
```

Comment:
>// this is permitted now, as it is just an assignment expression [](start = 12, length = 64)

Please add a test that verifies that the behavior of the compiled code is correct. Test SemanticModel as well.